jumanji/environments/routing/robot_warehouse/env.py

```diff
@@ -0,0 +1,544 @@
+        actions = utils.get_valid_actions(actions, state.action_mask)
+
+        # check for agent collisions
+        collisions = jax.vmap(functools.partial(utils.is_collision, grid))(
+            agents, actions
+        )
+        collision = jnp.sum(collisions) > 0
+
+        # update agents, shelves and grid
+        def update_state_scan(
+            carry_info: Tuple[chex.Array, chex.Array, chex.Array, int], action: int
+        ) -> Tuple[Tuple[chex.Array, chex.Array, chex.Array, int], None]:
+            grid, agents, shelves, agent_id = carry_info
+            grid, agents, shelves = self._update_state(
+                grid, agents, shelves, action, agent_id
+            )
+            return (grid, agents, shelves, agent_id + 1), None
+
+        (grid, agents, shelves, _), _ = jax.lax.scan(
+            update_state_scan, (grid, agents, shelves, 0), actions
+        )
+
+        # compute shared reward for all agents and update request queue
+        # if a requested shelf has been successfully delivered to the goal
+        reward = jnp.array(0, dtype=jnp.float32)
+
+        def update_reward_and_request_queue_scan(
+            carry_info: Tuple[
+                chex.PRNGKey, chex.Array, chex.Array, chex.Array, chex.Array
+            ],
+            goal: chex.Array,
+        ) -> Tuple[
+            Tuple[chex.PRNGKey, chex.Array, chex.Array, chex.Array, chex.Array], None
+        ]:
+            key, reward, request_queue, grid, shelves = carry_info
+            (
+                key,
+                reward,
+                request_queue,
+                shelves,
+            ) = self._update_reward_and_request_queue(
+                key, reward, request_queue, grid, shelves, goal
+            )
+            carry_info = (key, reward, request_queue, grid, shelves)
+            return carry_info, None
+
+        update_info, _ = jax.lax.scan(
+            update_reward_and_request_queue_scan,
+            (key, reward, request_queue, grid, shelves),
+            self.goals,
+        )
+        key, reward, request_queue, grid, shelves = update_info
+
+        # construct timestep and check environment termination
+        steps = state.step_count + 1
+        horizon_reached = steps >= self.time_limit
+        done = collision | horizon_reached
+
+        # compute next observation
+        agents_view = self._make_observations(grid, agents, shelves)
+        action_mask = utils.compute_action_mask(grid, agents)
+        next_observation = Observation(
+            agents_view=agents_view,
+            action_mask=action_mask,
+            step_count=steps,
+        )
+
+        timestep: TimeStep[chex.Array] = jax.lax.cond(
+            done,
+            # this really should be a truncation, but jumanji training doesn't support it
+            lambda _: termination(reward=reward, observation=next_observation),
+            lambda _: transition(reward=reward, observation=next_observation),
+            None,
+        )
```

Comment:
To be consistent with the other environments:
```suggestion
        timestep: = jax.lax.cond(
            done,
            termination,
            transition,
            reward,
            next_observation,
        )
```